lib/semantics/semantics.cc

```diff
@@ -91,30 +99,13 @@ const Scope &SemanticsContext::FindScope(
 }
 
 bool Semantics::Perform() {
-  ValidateLabels(context_.messages(), program_);
-  if (AnyFatalError()) {
-    return false;
-  }
-  parser::CanonicalizeDo(program_);
-  ResolveNames(context_, program_);
-  if (AnyFatalError()) {
-    return false;
-  }
-  RewriteParseTree(context_, program_);
-  if (AnyFatalError()) {
-    return false;
-  }
-  StatementSemanticsPass1{context_}.Walk(program_);
-  if (AnyFatalError()) {
-    return false;
-  }
-  StatementSemanticsPass2{context_}.Walk(program_);
-  if (AnyFatalError()) {
-    return false;
-  }
-  ModFileWriter writer{context_};
-  writer.WriteAll();
-  return !AnyFatalError();
+  return ValidateLabels(context_.messages(), program_) &&
+      parser::CanonicalizeDo(program_) &&  // force line break
```

Comment:
The comment is there to prevent clang-format from joining this line with the next.
Is there a better way to achieve that?

Reply:
You can also use `// clang-format off` just before and re-activate formatting with `// clang-format on`. But I think your comment is better because formatting is still applied inside your line.

Reply:
`clang-format off` is a good option for a block of code, but—I agree with Jean—this comment is a better way to split a line and still get some formatting.

Reply:
It's not the end of the world if the calls don't each occupy their own line.  But you might also try giving `ResolveNames` a longer name so it doesn't fit.

Or use template magic to build a tuple of lambdas from an initializer-list and call its members in order until one fails; the constructor call's items could have a trailing comma, like common::visitors, and clang-format might keep it prettier.  But that cure seems worse that the original problem.